xnmt/encoder.py

```diff
@@ -36,17 +36,27 @@ def transduce(self, embed_sent):
     """
     raise NotImplementedError('Unimplemented transduce for class:', self.__class__.__name__)
 
+  def get_final_state(self):
+    """ Return the state that represents the transduced sequence """
+    return None
```

Comment:
It means I need to implement this method to all the subclass of the Encoder. There are some encoders that are not used by the translator (only retriever). And get_final_state() is maybe confusing there. So instead I handle in the SoftmaxMLPDecoder, because it is used only in the translator and will throw an Error if the final_state is None.